<launch>
    <!-- Define arguments with default values -->
    <arg name="speed" default="0.3" />
    <arg name="min_dist_threshold" default="0.4" />
    <arg name="sensor_angles" default="[-150, -80, -20, -10, 10, 20, 80, 150]" />
    <arg name="dodge_angle_range" default="45" />

    <!-- Launch the mercator_rw_node with the specified parameters -->
    <node pkg="mercator_sensor_fusion_ros_pkg" type="mercator_rw" name="mercator_rw" output="screen">
        <param name="speed" value="$(arg speed)" />
        <param name="min_dist_threshold" value="$(arg min_dist_threshold)" />
        <param name="sensor_angles" value="$(arg sensor_angles)" />
        <param name="dodge_angle_range" value="$(arg dodge_angle_range)" />
    </node>
</launch>
